clientconn.go

```diff
@@ -435,7 +437,19 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 	if cc.dopts.copts.Dialer == nil {
 		cc.dopts.copts.Dialer = newProxyDialer(
 			func(ctx context.Context, addr string) (net.Conn, error) {
-				return dialContext(ctx, "tcp", addr)
+				network := "tcp"
+				p := regexp.MustCompile(`^[a-z]+://`)
```

Comment:
Make this static so we don't re-compile the regex in every dial?

Also, loose this to `^[a-z]+:/` so that `unix:/addr` would also work?

We would like this to work with `unix:addr` as well.
But `url.Parse` doesn't parse `"example.com:443"` correctly. It also fails to parse `"192.168.0.1:443"`

Reply:
`url.Parse` also fails on `unix:///tmp/etcd-unix-so-325611655`